addons/web/static/src/core/user.js

```diff
@@ -88,7 +89,7 @@ export function _makeUser(session) {
             return Object.assign({}, context, { uid: this.userId });
         },
         get lang() {
-            return this.context.lang;
+            return pyToJsLocale(this.context.lang);
```

Comment:
can't we find a way to compute this just once?

Reply:
Of course, but then `context.lang` can't change. I don't know of any use case where it is updated, though. If you think it's reasonable, I could store the result of `pyToJsLocale(this.context.lang)` in `_makeUser`'s closure :+1:

Reply:
On second thought, we could format user_context.lang before sending it to the client side

Reply:
@ged-odoo In the end, changing the context is a bad idea, it created way more diff and extra complexity (and I couldn't get the runbot green). I went back to my first solution, which is just formatting the language on the user service. Regarding your initial comment, I put the formatted language as part of the closure to avoid computing it every time 👍